### PR TITLE
Add instructions for building dotnet 6.0 from source

### PR DESCRIPTION
**Pull request by lbussell**
See https://github.com/dotnet/source-build/issues/2602 for more information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,101 @@
...
+    dotnet --info
+    ```
+
+## .NET 5.0 and .NET Core 3.1
+To build older versions of the .NET SDK from source, pick a specific Git tag with your desired version, or use a release branch to build the latest servicing release of that version. Refer to the tag/branch's README for build instructions:
+
+* [`release/5.0`](https://github.com/dotnet/source-build/tree/release/5.0)
+* [`release/3.1`](https://github.com/dotnet/source-build/tree/release/3.1)
+
+The scripts are written for Bash and supported on macOS and Linux.
```
Comment (MichaelSimons) [inline README.md:81]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,48 +1,101 @@
...
 
-* .NET Core is composed of many repositories that need to be built at a specific combination of commits.
+* .NET is composed of many repositories that need to be built at a specific combination of commits.
 * Each repository's build output needs to flow into the next repository's build.
-* By default, most .NET Core repositories download prebuilt binary dependencies from online sources. These are forbidden by typical Linux distribution rules, and interfere with build output flow.
-* Nearly all .NET Core repositories require the .NET Core SDK to build. This is a circular dependency, which presents a bootstrapping problem.
+* By default, most .NET repositories download prebuilt binary dependencies from online sources. These are forbidden by typical Linux distribution rules, and interfere with build output flow.
+* Nearly all .NET repositories require the .NET SDK to build. This is a circular dependency, which presents a bootstrapping problem.
 
 The source-build repository contains scripts and build logic to help Linux distribution maintainers address these challenges.
```
Comment (MichaelSimons) [inline README.md:100]:
```suggestion
.NET source-build contains scripts and build logic to help Linux distribution maintainers address these challenges.
```

File: README.md [hunk 3]
```diff
@@ -1,48 +1,101 @@
...
-```console
-./build.sh
-```
+    This builds the entire .NET SDK from source. The resulting SDK is placed at `artifacts/x64/Release/dotnet-sdk-6.0.100-fedora.33-x64.tar.gz`.
 
-Once the build is successful, the outputs are placed in `artifacts/packages/Debug/`.
+5. (Optional) Unpack and install the .NET SDK
+
+    ```bash
+    mkdir -p $HOME/dotnet && tar zxf artifacts/x64/Release/dotnet-sdk-6.0.100-fedora.33-x64.tar.gz -C $HOME/dotnet
```
Comment (MichaelSimons) [inline README.md:67]:
```suggestion
    mkdir -p $HOME/dotnet
    tar zxf artifacts/x64/Release/dotnet-sdk-6.0.100-fedora.33-x64.tar.gz -C $HOME/dotnet
```

File: README.md [hunk 4]
```diff
@@ -1,48 +1,101 @@
...
 
-The current branch is a work in progress that produces several .NET SDK components for [Arcade-powered source-build](Documentation/planning/arcade-powered-source-build). Once Arcade-powered source-build is complete, it will allow building a .NET SDK from source directly from the [dotnet/installer](https://github.com/dotnet/installer) repository.
+The dependencies for building .NET 6.0 from source can be found [here](https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md).
 
-## Using this repository
+### Building
 
-The scripts are written for Bash and supported on macOS and Linux.
+.NET 6.0 is built from source using the [dotnet/installer](https://github.com/dotnet/installer) repo.
+...
```
Comment (MichaelSimons) [inline README.md:16]:
```suggestion

```

File: README.md [hunk 5]
```diff
@@ -1,48 +1,101 @@
...
+For example, in Fedora 33:
```
Comment (MichaelSimons) [inline README.md:17]:
```suggestion
The steps to build vary slightly depending on your distro.  The following set of instructions walk through how to build on Fedora.33.  
```

File: README.md [hunk 6]
```diff
@@ -1,48 +1,104 @@
...
+
+3. Create a .NET source tarball.
+
+   ```bash
+   ./build.sh /p:ArcadeBuildTarball=true /p:TarballDir=/path/to/place/complete/dotnet/sources
+   ```
+
+   This fetches the complete .NET source code and creates a tarball at `artifacts/packages/<Release|Debug>/Shipping/`.
+   The extracted source code is also placed at `/path/to/place/complete/dotnet/sources`.
+   The source directory should be outside (and not somewhere under) this repository.
```
Comment (dseefeld) [inline README.md:34]:
```suggestion
   The source directory should be outside (and not somewhere under) the installer directory.
```

File: README.md [hunk 7]
```diff
@@ -1,48 +1,104 @@
...
+
+3. Prep the source to build on your distro.
+
+    ```bash
+    cd /path/to/complete/dotnet/sources
+    ./prep.sh
+    ```
+
+    This downloads a .NET SDK and a number of .NET packages and other prebuilts needed to build .NET from source.
```
Comment (dseefeld) [inline README.md:42]:
```suggestion
    This downloads a .NET SDK and a number of .NET packages needed to build .NET from source.
```

File: README.md [hunk 8]
```diff
@@ -1,50 +1,105 @@
...
+
+    > On Linux distros other than Fedora 33, an additional bootstrapping step is required.  After running `prep.sh` above, run the following:
+    >
+    > ```bash
+    > tar xf ../packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz
+    > sed -i -E 's|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.1</|> <MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
+    > sed -i -E 's|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.1</|> <MicrosoftNETHostModelVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
```
Comment (omajid) [inline README.md:49]: I think there's a few extra characters here:

```suggestion
    > sed -i -E 's|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
    > sed -i -E 's|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
```

File: README.md [hunk 9]
```diff
@@ -1,50 +1,105 @@
-# .NET Core Source-Build
+# .NET Source-Build
 
-[![Build Status](https://dev.azure.com/dnceng/internal/_apis/build/status/dotnet/source-build/source-build-rolling-CI?branchName=master)](https://dev.azure.com/dnceng/internal/_build/latest?definitionId=114&branchName=master)
 [![Join the chat at https://gitter.im/dotnet/source-build](https://badges.gitter.im/dotnet/source-build.svg)](https://gitter.im/dotnet/source-build?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
 This repository helps .NET SDK and .NET Runtime package maintainers comply with common Linux distribution guidelines.
```
Comment (MichaelSimons) [inline README.md:5]:
```suggestion
This repo is the starting point for building .NET from source.
```

File: README.md [hunk 10]
```diff
@@ -1,50 +1,105 @@
...
 
-The current branch is a work in progress that produces several .NET SDK components for [Arcade-powered source-build](Documentation/planning/arcade-powered-source-build). Once Arcade-powered source-build is complete, it will allow building a .NET SDK from source directly from the [dotnet/installer](https://github.com/dotnet/installer) repository.
+The dependencies for building .NET 6.0 from source can be found [here](https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md).
 
-## Using this repository
+### Building
 
-The scripts are written for Bash and supported on macOS and Linux.
+.NET 6.0 is built from source using the [dotnet/installer](https://github.com/dotnet/installer) repo.
+The steps to build vary slightly depending on your distro.  The following set of instructions walk through how to build on Fedora.33.
```
Comment (dleeapho) [inline README.md:16]: Typo: period after 'Fedora'
```suggestion
The steps to build vary slightly depending on your distro.  The following set of instructions walk through how to build on Fedora 33.  
```

**Review by dleeapho — APPROVED** [review]
LGTM.

File: README.md [hunk 11]
```diff
@@ -1,50 +1,105 @@
...
+
+    This downloads a .NET SDK and a number of .NET packages needed to build .NET from source.
+
+    > On Linux distros other than Fedora 33, an additional bootstrapping step is required.  After running `prep.sh` above, run the following:
+    >
+    > ```bash
+    > tar xf ../packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz
+    > sed -i -E 's|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
+    > sed -i -E 's|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
+    > tar czf ../packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz *
```
Comment (dseefeld) [inline README.md:50]:
```suggestion
    > mkdir ./privateSourceBuiltArtifacts
    > cd ./privateSourceBuiltArtifacts
    > tar xf ../packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz
    > sed -i -E 's|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelPackageVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
    > sed -i -E 's|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.1</|<MicrosoftNETHostModelVersion>6.0.0-rtm.21521.4</|' PackageVersions.props
    > tar czf ../packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz *
    > cd ..
    > rm -r ./privateSourceBuiltArtifacts
```

Comment (dseefeld) [reply README.md:50]: The commands worked fine, but assumed you were in an empty directory that is a peer to `./packages`.  I added the directory commands before and after.

File: README.md [hunk 12]
```diff
@@ -1,50 +1,109 @@
...
+To build older versions of the .NET SDK from source, pick a specific Git tag with your desired version, or use a release branch to build the latest servicing release of that version. Refer to the tag/branch's README for build instructions:
+
+* [`release/5.0`](https://github.com/dotnet/source-build/tree/release/5.0)
+* [`release/3.1`](https://github.com/dotnet/source-build/tree/release/3.1)
+
+
+> The source-build repository doesn't currently support Windows. See [source-build#1190](https://github.com/dotnet/source-build/issues/1190).
 
-## Goals
+# Source build goals
```
Comment (richlander) [inline README.md:91]: This should probably be an H2.
```suggestion
## Source build goals
```

File: README.md [hunk 13]
```diff
@@ -1,50 +1,109 @@
...
+# .NET Source-Build
 
-[![Build Status](https://dev.azure.com/dnceng/internal/_apis/build/status/dotnet/source-build/source-build-rolling-CI?branchName=master)](https://dev.azure.com/dnceng/internal/_build/latest?definitionId=114&branchName=master)
 [![Join the chat at https://gitter.im/dotnet/source-build](https://badges.gitter.im/dotnet/source-build.svg)](https://gitter.im/dotnet/source-build?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-This repository helps .NET SDK and .NET Runtime package maintainers comply with common Linux distribution guidelines.
+This repo is the starting point for building .NET from source.
 
-To build the full .NET SDK from source, pick a specific Git tag with your desired version, or use a release branch to build the latest servicing release of that version. Refer to the tag/branch's README for build instructions:
+## .NET 6.0
```
Comment (richlander) [inline README.md:7]: I think we should turf this head then make the following headings H2s.

File: README.md [hunk 14]
```diff
@@ -1,50 +1,109 @@
-# .NET Core Source-Build
+# .NET Source-Build
 
-[![Build Status](https://dev.azure.com/dnceng/internal/_apis/build/status/dotnet/source-build/source-build-rolling-CI?branchName=master)](https://dev.azure.com/dnceng/internal/_build/latest?definitionId=114&branchName=master)
 [![Join the chat at https://gitter.im/dotnet/source-build](https://badges.gitter.im/dotnet/source-build.svg)](https://gitter.im/dotnet/source-build?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-This repository helps .NET SDK and .NET Runtime package maintainers comply with common Linux distribution guidelines.
+This repo is the starting point for building .NET from source.
```
Comment (richlander) [inline README.md:5]:
```suggestion
This repo is the starting point for building .NET 6 from source. Instructions for building other .NET versions are provided near the end of this document.
```

**Review by richlander — APPROVED** [review]

**Comment by rbhanda** [comment]
I have updated the tag on installer repo to https://github.com/dotnet/installer/releases/tag/v6.0.100

**Pull request merged**